hall-of-fame-service/src/main/java/hcmus/alumni/halloffame/repository/HallOfFameRepository.java

```diff
@@ -0,0 +1,46 @@
+package hcmus.alumni.halloffame.repository;
+
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+import org.springframework.transaction.annotation.Transactional;
+
+import hcmus.alumni.halloffame.dto.IHallOfFameDto;
+import hcmus.alumni.halloffame.model.HallOfFameModel;
+
+
+public interface HallOfFameRepository extends JpaRepository<HallOfFameModel, String> {
+	Optional<IHallOfFameDto> findHallOfFameById(String id);
+	
+	Optional<HallOfFameModel> findById(String id);
+	
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s WHERE s.id != 4 AND n.title like %:title%")
+	Page<IHallOfFameDto> searchHof(String title, Pageable pageable);
+	
+	@Query("SELECT n FROM HallOfFameModel n JOIN n.status s WHERE s.id = :statusId AND n.title like %:title%")
+	Page<IHallOfFameDto> searchHofByStatus(String title, Integer statusId, Pageable pageable);
```

Comment:
thêm search theo khoa với khoá
[tham khảo](https://stackoverflow.com/questions/70165854/spring-data-jpa-how-to-form-a-conditional-query)